r/tests/testthat/test-data-type.R

```diff
@@ -395,6 +395,18 @@ test_that("decimal type and validation", {
   expect_error(decimal(4, NA), '"scale" must be an integer')
 
   expect_r6_class(decimal(4, 2), "Decimal128Type")
+
+  expect_error(decimal128())
+  expect_error(decimal128("four"), '"precision" must be an integer')
+  expect_error(decimal128(4))
+  expect_error(decimal128(4, "two"), '"scale" must be an integer')
+  expect_error(decimal128(NA, 2), '"precision" must be an integer')
+  expect_error(decimal128(0, 2), "Invalid: Decimal precision out of range: 0")
+  expect_error(decimal128(100, 2), "Invalid: Decimal precision out of range: 100")
+  expect_error(decimal128(4, NA), '"scale" must be an integer')
+
+  expect_r6_class(decimal128(4, 2), "Decimal128Type")
```

Comment:
This is super super minor, but maybe we should move this up above the `expect_errors()`. We aren't strict about this, but _for the most part_ the pattern we use is test the base cases, test edge cases, test errors (though I'm sure there are counter examples!)

Reply:
Done